manticore/__main__.py

```diff
@@ -79,51 +78,38 @@ def parse_arguments():
 def main():
     args = parse_arguments()
 
-    m = Manticore(args.argv[0], args.argv[1:], disasm=args.disasm)
-    m.policy = args.policy
-    m.args = args
+    log.set_verbosity(args.v)
+
+
+    #TODO 
+    initial_state = make_initial_state(args.argv[0], argv=args.argv[1:],  disasm=args.disasm)
+    m = Manticore(initial_state, workspace_url=args.workspace,  policy=args.policy)
+
+    #Fixme remove all this...
+    m.coverage_file = args.coverage
 
     if args.data:
         m.concrete_data = args.data
 
-    if args.workspace:
-        m.workspace = args.workspace
-
     if args.profile:
         m.should_profile = args.profile
 
-    if args.dumpafter != 0:
-        m.dumpafter = args.dumpafter
-
-    if args.maxstorage != 0:
-        m.maxstorage = args.maxstorage
-
-    if args.maxstates != 0:
-        m.maxstates = args.maxstates
-
-    if args.coverage:
-        m.coverage_file = args.coverage
-
     if args.names is not None:
         m.apply_model_hooks(args.names)
 
     if args.env:
         for entry in args.env:
             name, val = entry[0].split('=')
-            m.env_add(name, val)
```

Comment:
i think both of these will crash, state doesn't have either env_add or add_symbolic_file. these would fit perfectly on the platform though, which we can call from the init hook

Reply:
actually env_add will be challenging, because Linux needs that when loading the binary

Reply:
m.initial_state.platform.env_add(name, val) ?